include/gsl/span_ext

```diff
@@ -1,198 +1,170 @@
-{
-    return std::lexicographical_compare(l.begin(), l.end(), r.begin(), r.end());
-}
-
-template <class ElementType, std::size_t Extent>
-constexpr bool operator<=(span<ElementType, Extent> l, span<ElementType, Extent> r)
-{
-    return !(l > r);
-}
-
-template <class ElementType, std::size_t Extent>
-constexpr bool operator>(span<ElementType, Extent> l, span<ElementType, Extent> r)
-{
-    return r < l;
-}
-
-template <class ElementType, std::size_t Extent>
-constexpr bool operator>=(span<ElementType, Extent> l, span<ElementType, Extent> r)
-{
-    return !(l < r);
-}
-
-//
-// make_span() - Utility functions for creating spans
-//
-template <class ElementType>
-constexpr span<ElementType> make_span(ElementType* ptr, typename span<ElementType>::size_type count)
-{
-    return span<ElementType>(ptr, count);
-}
-
-template <class ElementType>
-constexpr span<ElementType> make_span(ElementType* firstElem, ElementType* lastElem)
-{
-    return span<ElementType>(firstElem, lastElem);
-}
-
-template <class ElementType, std::size_t N>
-constexpr span<ElementType, N> make_span(ElementType (&arr)[N]) noexcept
-{
-    return span<ElementType, N>(arr);
-}
-
-template <class Container>
-constexpr span<typename Container::value_type> make_span(Container& cont)
-{
-    return span<typename Container::value_type>(cont);
-}
-
-template <class Container>
-constexpr span<const typename Container::value_type> make_span(const Container& cont)
-{
-    return span<const typename Container::value_type>(cont);
-}
-
-template <class Ptr>
-constexpr span<typename Ptr::element_type> make_span(Ptr& cont, std::size_t count)
-{
-    return span<typename Ptr::element_type>(cont, count);
-}
-
-template <class Ptr>
-constexpr span<typename Ptr::element_type> make_span(Ptr& cont)
-{
-    return span<typename Ptr::element_type>(cont);
-}
-
-// Specialization of gsl::at for span
-template <class ElementType, std::size_t Extent>
-constexpr ElementType& at(span<ElementType, Extent> s, index i)
-{
-    // No bounds checking here because it is done in span::operator[] called below
-    Ensures(i >= 0);
-    return s[narrow_cast<std::size_t>(i)];
-}
-
-// [span.obs] Free observer functions
-template <class ElementType, std::size_t Extent>
-constexpr std::ptrdiff_t ssize(const span<ElementType, Extent>& s) noexcept
-{
-    return static_cast<std::ptrdiff_t>(s.size());
-}
-
-// [span.iter] Free functions for begin/end functions
-template <class ElementType, std::size_t Extent>
-constexpr typename span<ElementType, Extent>::iterator
-begin(const span<ElementType, Extent>& s) noexcept
-{
-    return s.begin();
-}
-
-template <class ElementType, std::size_t Extent = dynamic_extent>
-constexpr typename span<ElementType, Extent>::iterator
-end(const span<ElementType, Extent>& s) noexcept
-{
-    return s.end();
-}
-
-template <class ElementType, std::size_t Extent>
-constexpr typename span<ElementType, Extent>::const_iterator
-cbegin(const span<ElementType, Extent>& s) noexcept
```

Comment:
IIUC, these functions exist to enable `meow(some_span)` syntax in cases where `meow(some_std_span)` would call `std::meow`. If that's the case, this should still exist and should have been returning `s,begin()` (Yes, member and non-member `cbegin` for `std::span` had different behavior. This was much of the motivation for the change.)